### ISSUE TITLE
Binarybuild installation fails on macOS

### ISSUE THREAD

**Issue by evetion** [JuliaGeo/GDAL.jl#51]
GDAL doesn't install on macOS, failing on the building part. 

```julia
julia> Pkg.build("GDAL")
INFO: Building CodecZlib
INFO: Building Proj4
INFO: Building LibGEOS
INFO: Building GDAL
==================================================================================================[ ERROR: GDAL ]===================================================================================================

LoadError: BinaryProvider.LibraryProduct(nothing, String["libgdal"], :libgdal, "Prefix(/Users/epta/.julia/v0.6/GDAL/deps/usr)") is not satisfied, cannot generate deps.jl!
while loading /Users/epta/.julia/v0.6/GDAL/deps/build.jl, in expression starting on line 56

====================================================================================================================================================================================================================

```

The problem lies in the binary build, the `libgdal.dylib` cannot be loaded because its depencies cannot be found. The following lines prevent it for Linux, but not for macOS: https://github.com/JuliaGeo/GDAL.jl/blob/master/deps/build.jl#L46

```julia
julia> Libdl.dlopen("libgdal")
ERROR: could not load library "libgdal"
dlopen(libgdal.dylib, 1): Library not loaded: @rpath/libproj.12.dylib
  Referenced from: /Users/epta/.julia/v0.6/GDAL/deps/usr/lib/libgdal.20.dylib
  Reason: image not found
Stacktrace:
 [1] dlopen(::String, ::UInt32) at ./libdl.jl:97 (repeats 2 times)
```

**Comment by visr** [comment]
Hmm I wonder why it's working on OSX Travis but not for you locally. In any case I'm updating the builders to run with the latest BinaryBuilder which might bring along some rpath related fixes. Hope that will help here.

**Comment by evetion** [comment]
This is fixed for me by #52
